gcp-observability/src/main/java/io/grpc/gcp/observability/ObservabilityConfigImpl.java

```diff
@@ -74,95 +87,128 @@ void parse(String config) throws IOException {
-          builder.put(entry.getKey(), (String) entry.getValue());
-        }
-        customTags = builder.build();
+            entry.getValue() instanceof String,
+            "'labels' needs to be a map of <string, string>");
+        builder.put(entry.getKey(), (String) entry.getValue());
       }
+      customTags = builder.build();
     }
   }
 
-  private EventType convertEventType(String val) {
-    switch (val) {
-      case "GRPC_CALL_UNKNOWN":
-        return EventType.GRPC_CALL_UNKNOWN;
-      case "GRPC_CALL_REQUEST_HEADER":
-        return EventType.GRPC_CALL_REQUEST_HEADER;
-      case "GRPC_CALL_RESPONSE_HEADER":
-        return EventType.GRPC_CALL_RESPONSE_HEADER;
-      case "GRPC_CALL_REQUEST_MESSAGE":
-        return EventType.GRPC_CALL_REQUEST_MESSAGE;
-      case "GRPC_CALL_RESPONSE_MESSAGE":
-        return EventType.GRPC_CALL_RESPONSE_MESSAGE;
-      case "GRPC_CALL_TRAILER":
-        return EventType.GRPC_CALL_TRAILER;
-      case "GRPC_CALL_HALF_CLOSE":
-        return EventType.GRPC_CALL_HALF_CLOSE;
-      case "GRPC_CALL_CANCEL":
-        return EventType.GRPC_CALL_CANCEL;
-      default:
-        throw new IllegalArgumentException("Unknown event type value:" + val);
+  private List<LogFilter> parseRpcEvents(List<?> rpcEvents) {
+    List<Map<String, ?>> jsonRpcEvents = JsonUtil.checkObjectList(rpcEvents);
+    ImmutableList.Builder<LogFilter> rpcEventsListBuilder =
+        new ImmutableList.Builder<>();
+    for (Map<String, ?> jsonClientRpcEvent : jsonRpcEvents) {
+      rpcEventsListBuilder.add(parseJsonLogFilter(jsonClientRpcEvent));
     }
+    return rpcEventsListBuilder.build();
   }
 
+
   private LogFilter parseJsonLogFilter(Map<String, ?> logFilterMap) {
-    return new LogFilter(JsonUtil.getString(logFilterMap, "pattern"),
-        JsonUtil.getNumberAsInteger(logFilterMap, "header_bytes"),
-        JsonUtil.getNumberAsInteger(logFilterMap, "message_bytes"));
+    Boolean exclude = JsonUtil.getBoolean(logFilterMap, "exclude");
+    Boolean global = false;
+    ImmutableList.Builder<String> patternListBuilder =
+        new ImmutableList.Builder<>();
+    ImmutableSet.Builder<String> servicesSetBuilder =
+        new ImmutableSet.Builder<>();
+    ImmutableSet.Builder<String> methodsSetBuilder =
+        new ImmutableSet.Builder<>();
+    List<?> methodsList = JsonUtil.getList(logFilterMap, "methods");
+    if (methodsList != null) {
+      for (Object pattern : methodsList) {
+        String methodOrServicePattern = String.valueOf(pattern);
+        if (methodOrServicePattern != null) {
+          if (methodOrServicePattern.equals("*")) {
+            if (exclude != null) {
+              checkArgument(
+                  !exclude,
+                  "cannot have 'exclude' and '*' wildcard in the same filter");
+            }
+            global = true;
+          } else {
+            checkArgument(
+                METHOD_NAME_REGEX.matcher(methodOrServicePattern).matches(),
```

Comment:
Wonder if you enhance your `METHOD_NAME_REGEX` to also include just `"*"` this code blob will be simpler/ more readable.

```
 checkArgument(METHOD_NAME_REGEX....);
 if ("*".equals(methodOrServicePattern)) {
    checkArgument(!exclude, ...);
    ...
 } else {
    if (methodOrServicePattern.endsWith("/*")) {
       ...
    } else {
       ...
    }
 }
```

Reply:
Done. PTAL

Reply:
nit: Since the regex matcher has done all the hard/expensive work and you are using capturing groups in your regex how about leveraging it:

```
Matcher matcher = METHOD_NAME_REGEX.matcher(methodOrServicePattern);
checkArgument(matcher.matches(), ...);
if (methodOrServicePattern.equals("*")) {
        checkArgument(!exclude, "cannot have 'exclude' and '*' wildcard in the same filter");
        globalFilter = true;
      } else if ("*".equals(matcher.group(5))) { // 5 for method name ?
        String service = matcher.group(4);  // 4 is for service name
```

Reply:
Done.